### PR TITLE
Lectern Client 2.X migration

### PR DESCRIPTION
**Pull request by leoraba**
- [x] Replace Lectern client from `@overturebio-stack/lectern-client": "^1.5.0"` to `@overture-stack/lectern-client": "2.0.0-beta.2`
- [x] Replace legacy types and objects to new ones
- [x] Use `parse` and `validate` functions provided by `lectern-client@2.x`
- [x] Update/Add Unit Tests

### CODE REVIEW

File: packages/data-model/src/models/dictionaries.ts [hunk 1]
```diff
@@ -1,6 +1,7 @@
 import { jsonb, pgTable, serial, timestamp, varchar } from 'drizzle-orm/pg-core';
 
-import { SchemaDefinition } from '@overturebio-stack/lectern-client/lib/schema-entities.js';
+import { Schema } from '@overture-stack/lectern-client';
+interface SchemaDefinition extends Schema {}
```
Comment (leoraba) [inline packages/data-model/src/models/dictionaries.ts:4]: Wrapping the `Schema` zod type is needed in order to be used as a column type definition on a Drizzle object.

File: packages/data-provider/src/utils/fileUtils.ts [hunk 2]
```diff
@@ -30,33 +34,39 @@ export const readHeaders = async (file: Express.Multer.File) => {
...
+	fileName: string,
+	schema: Schema,
+): Promise<{ records: DataRecord[]; errors?: ParseSchemaError[] }> => {
 	const contents = await fsPromises.readFile(fileName, 'utf-8');
 	const arr = parseTsvToJson(contents);
-	return arr;
+	const parseSchemaResult = parse.parseSchemaValues(arr, schema);
+	if (parseSchemaResult.success) {
+		return { records: parseSchemaResult.data.records };
+	}
```
Comment (leoraba) [inline packages/data-provider/src/utils/fileUtils.ts:50]: this is a function provided by lectern v2.x to parse raw string data into it's corresponding data type provided by the Dictionary schema

**Comment by leoraba** [comment]
Comparison of the data types and functions

| Lectern-client 1.x | Lectern-client 2.x |
| - | - |
| <pre>import { <strong>SchemaData</strong> } from '@overturebio-stack/lectern-client/lib/schema-entities.js';</pre> | <pre>import { DataRecord } from '@overture-stack/lectern-client';<br /><br /><strong>Array\<DataRecord\></strong></pre> |
| <pre> import { type <strong>DataRecord</strong> } from '@overturebio-stack/lectern-client/lib/schema-entities.js';</pre> | <pre>import { type <strong>DataRecord</strong> } from '@overture-stack/lectern-client';</pre> |
| <pre>import { <strong>SchemaValidationError</strong> } from '@overturebio-stack/lectern-client/lib/schema-entities.js';</pre>|<pre>import type { <strong>DictionaryValidationRecordErrorDetails</strong> } from '@overture-stack/lectern-client';</pre>|
|<pre>import { entities as dictionaryEntities } from '@overturebio-stack/lectern-client';<br /><br /><strong>dictionaryEntities.SchemasDictionary</strong></pre> | <pre>import { Dictionary as <strong>SchemasDictionary</strong> } from '@overture-stack/lectern-client';</pre> |
|<pre>import { restClient as <strong>dictionaryRestClient</strong> } from '@overturebio-stack/lectern-client';</pre>|<pre>import { rest as <strong>dictionaryRestClient</strong> } from '@overture-stack/lectern-client';</pre>|
|<pre>import { type <strong>BatchProcessingResult</strong> } from '@overturebio-stack/lectern-client/lib/schema-entities.js';<br /><br />Record<string, <strong>BatchProcessingResult</strong>><pre>|<pre>import { DictionaryValidationError, TestResult } from '@overture-stack/lectern-client';<br /><br />TestResult<<strong>DictionaryValidationError[]</strong>></pre>|
|<pre>import { <strong>processSchemas</strong> } from '@overturebio-stack/lectern-client/lib/schema-functions.js';<br /><br /><strong>processSchema</strong>(schemasDictionary, schemasData);</pre>|<pre>import { <strong>validate</strong> } from '@overture-stack/lectern-client';<br /><br /><strong>validate.validateDictionary</strong>(schemasData, schemasDictionary);</pre>|
|<pre>import { <strong>SchemaValidationError</strong> } from '@overturebio-stack/lectern-client/lib/schema-entities.js';</pre>|<pre>import { <strong>DictionaryValidationRecordErrorDetails</strong> } from '@overture-stack/lectern-client';</pre>|
|<pre>import { <strong>SchemaDefinition</strong> } from '@overture-stack/lectern-client';</pre>|<pre>import { <strong>Schema</strong> } from '@overture-stack/lectern-client';<br /><br />interface <strong>SchemaDefinition</strong> extends Schema {}</pre>|

File: packages/data-model/package.json [hunk 3]
```diff
@@ -39,7 +39,7 @@
 		"tsx": "^4.16.2"
 	},
 	"dependencies": {
-		"@overturebio-stack/lectern-client": "^1.5.0",
```
Comment (justincorrigible) [inline packages/data-model/package.json:42]: 🎉

**Review by joneubank — APPROVED** [review]

**Pull request merged**